### PR TITLE
[msal-angular] Interceptor, guard, and samples to use active account

### PR DESCRIPTION
**Pull request by jo-arroyo**
This PR:
- Updates the MSAL interceptor to use the active account with `getActiveAccount`, with a fallback to the first account in `getAllAccounts`. Log messages added and unit tests updated for change.
- Updates MSAL guard to set active account with `setActiveAccount` after logging in with popup.
- Updates msal-angular v2 samples to also set active account after logging in with popup or calling `handleRedirectObservable`.
- Updates upgrade guide with account methods available and our recommendations for use.
- Updates `angular11-sample-app` readme with link to initialization doc.

### CODE REVIEW

**Comment by jasonnutter** [comment]
Should also use `setActiveAccount` in teh places in the samples where `handleRedirectObservable` is called.

File: lib/msal-angular/docs/v2-docs/1.x-2.x-upgrade-guide.md [hunk 1]
```diff
@@ -43,10 +43,14 @@ See the [updated sample](https://github.com/AzureAD/microsoft-authentication-lib
 
 ### API changes
 * The `acquireToken` and `login` methods now take different request objects as parameters. See the [msal.service.ts](https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/dev/lib/msal-angular/src/msal.service.ts) for details.
-* `getAccount()` is now `getAccountByUsername()`. There is also a `getAllAccounts()` method available.
 * Broadcast events now emit an `EventMessage` object, instead of just strings. See the [Angular sample](https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/dev/samples/msal-angular-v2-samples/angular10-sample-app/src/app/app.component.ts) for an example of how to implement.
 * Applications using `Redirect` methods must implement the `handleRedirectObservable` method (and have it run on every page load), which will capture the result of redirect operations. See the [Angular sample](https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/dev/samples/msal-angular-v2-samples/angular10-sample-app/src/app/home/home.component.ts) for an example of how to implement.
 
+### Accounts
+* `getAccount()` is now `getAccountByUsername()` and is available on the MSAL instance.
```
Comment (tnorling) [inline lib/msal-angular/docs/v2-docs/1.x-2.x-upgrade-guide.md:50]: We should be advertising `getAccountByLocalId` or `getAccountByHomeId` as the gold standard with `getAccountByUsername` being considered less reliable and for convenience only.

**Review by tnorling — APPROVED** [review]
One comment on messaging but otherwise looks good

File: lib/msal-angular/docs/v2-docs/1.x-2.x-upgrade-guide.md (hunk 1 above)
Comment (jo-arroyo) [reply lib/msal-angular/docs/v2-docs/1.x-2.x-upgrade-guide.md]: Thanks! I changed the upgrade guide to recommend that.

**Review by pkanher617 — APPROVED** [review]

**Review by jasonnutter — APPROVED** [review]

**Pull request merged**